SoftwareCo/SoftwareCo/Tracker/managers/HashManager.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Sodium;
+
+namespace SoftwareCo
+{
+    public class HashManager
+    {
+        public static String HashValue(string value, string dataType)
+        {
+            if (!CacheManager.HasJwt() || string.IsNullOrEmpty(value))
+            {
+                return "";
+            }
+
+            try
+            {
+                byte[] key = GenericHash.GenerateKey(); // 64 byte key
+                string hashedValue = Utilities.BinaryToHex(GenericHash.Hash(value, key, 128));
+                if (CacheManager.HasCachedValue(dataType, hashedValue))
```

Comment:
I think the logic is backward here.   Should be `if (CacheManager.DoesNotHaveCachedValue(dataType, hashedValue))`

where `CacheManager.DoesNotHaveCachedValue` == `!CacheManager.HasCachedValue`